library/init/meta/tactic.lean

```diff
@@ -550,14 +550,22 @@ meta constant set_basic_attribute (attr_name : name) (c_name : name) (persistent
 /-- `unset_attribute attr_name c_name` -/
 meta constant unset_attribute : name → name → tactic unit
 /-- `has_attribute attr_name c_name` succeeds if the declaration `decl_name`
-   has the attribute `attr_name`. The result is the priority. -/
-meta constant has_attribute : name → name → tactic nat
+   has the attribute `attr_name`. The result is the priority and whether or not
+   the attribute is persistent. -/
+meta constant has_attribute : name → name → tactic (bool × nat)
```

Comment:
This is not backwards compatible, since it changes the return type.

Reply:
Are we now ok with changing the signature?

Reply:
It's ok, it will only break one line in mathlib.

Reply:
The `has_attribute` function is pretty horrible to use on its own anyway, since it fails if the attribute is not present.  Everybody uses the mathlib function `has_attribute'` anyhow.

Reply:
This might be a good time to merge `has_attribute'` with `has_attribute`. How about that?